### PR TITLE
feat(secrets): Override secrets validation flag with tenant config

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/secrets/runner.py [hunk 1]
```diff
@@ -363,15 +363,25 @@ def save_secret_to_coordinator(
...
 
         if bc_integration.skip_download:
             logging.debug('Skipping secrets verification as flag skip-download was specified')
             return VerifySecretsResult.INSUFFICIENT_PARAMS
 
+        validate_secrets_tenant_config = bc_integration.customer_run_config_response is not None and \
+             bc_integration.customer_run_config_response.get('tenantConfig', {}).get('secretsValidate')
+
+        if validate_secrets_tenant_config is None and not convert_str_to_bool(os.getenv("CKV_VALIDATE_SECRETS", False)):
+            logging.debug('Secrets verification is off, enabled it via code configuration')
```
Comment (nimrodkor) [inline checkov/secrets/runner.py:378]:
```suggestion
            logging.debug('Secrets verification is off, enable it via code configuration screen')
```

File: checkov/secrets/runner.py [hunk 2]
```diff
@@ -363,15 +363,25 @@ def save_secret_to_coordinator(
 
     @time_it
     def verify_secrets(self, report: Report, enriched_secrets_s3_path: str) -> VerifySecretsResult:
-        if not bc_integration.bc_api_key or not convert_str_to_bool(os.getenv("CKV_VALIDATE_SECRETS", False)):
-            logging.debug(
-                'Secrets verification is off, enabled it via env var CKV_VALIDATE_SECRETS and provide an api key')
+        if not bc_integration.bc_api_key:
+            logging.debug('Unable to run secrets verification without a bc api key')
```
Comment (nimrodkor) [inline checkov/secrets/runner.py:367]:
```suggestion
            logging.debug('Secrets verification is available only with a valid API key')
```

File: checkov/secrets/runner.py [hunk 3]
```diff
@@ -363,15 +363,25 @@ def save_secret_to_coordinator(
...
 
+        validate_secrets_tenant_config = bc_integration.customer_run_config_response is not None and \
+             bc_integration.customer_run_config_response.get('tenantConfig', {}).get('secretsValidate')
+
+        if validate_secrets_tenant_config is None and not convert_str_to_bool(os.getenv("CKV_VALIDATE_SECRETS", False)):
+            logging.debug('Secrets verification is off, enabled it via code configuration')
+            return VerifySecretsResult.INSUFFICIENT_PARAMS
+
+        if validate_secrets_tenant_config is False:
+            logging.debug('Secrets verification is off, enabled it via code configuration')
```
Comment (nimrodkor) [inline checkov/secrets/runner.py:382]:
```suggestion
            logging.debug('Secrets verification is off, enable it via code configuration screen')
```

**Review by nimrodkor — APPROVED** [review]
Minor copywriting stuff

**Review by omryMen — APPROVED** [review]

**Review by marynaKK — APPROVED** [review]

**Pull request merged**